### PR TITLE
Added Integration tests for model_upload

### PR DESCRIPTION
**Pull request by mohami2000**
Added documentation for users as well

https://b.corp.google.com/issues/319686130

### CODE REVIEW

File: integration_tests/test_model_upload.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import os
+import tempfile
+import unittest
+
+from kagglehub import model_upload, models_helpers
+
+LICENSE_NAME = "MIT"
+OWNER_SLUG = "integrationtester"
```
Comment (rosbo) [inline integration_tests/test_model_upload.py:8]: For the creation case, let's create the model under the username of the person creating the test.

Comment (rosbo) [reply integration_tests/test_model_upload.py:8]: Using from kagglehub.config import get_kaggle_credentials

**Review by rosbo — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -107,6 +107,42 @@ hatch run test
...
+
+```sh
+# Run all tests
+hatch run integration-test
+
+# Run a single test file
+hatch run integration-test integration-test/test_<SOME_FILE>.py
+```
+
+To run integration tests on your local machine, you need to set up your Kaggle API credentials. There are two ways to do this:
```
Comment (rosbo) [inline README.md:120]: Does reading the credentials from `kaggle.json` works for these tests locally? 

https://github.com/Kaggle/kagglehub?tab=readme-ov-file#option-3-read-credentials-from-kagglejson

File: README.md [hunk 3]
```diff
@@ -107,6 +107,42 @@ hatch run test
...
+
+Using Environment Variables:
```
Comment (rosbo) [inline README.md:122]: To avoid duplication, I would simply link to `#option-2-read-credentials-from-environment-variables` (section above).

**Review by rosbo — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mohami2000) [reply README.md]: seems like it worked for me. adding it to the doc

File: README.md [hunk 4]
```diff
@@ -107,6 +107,24 @@ hatch run test
 hatch run test tests/test_<SOME_FILE>.py
 ```
 
+### Integration Tests
+
+To run integration tests on your local machine, you need to set up your Kaggle API credentials. You can do this in one of these two ways described in the earlier sections of this document. Refer to the sections: 
+Option 2: Read Credentials from Environment Variables
```
Comment (rosbo) [inline README.md:113]: can you link to these section using the anchors?

```
[...] Two ways [...]:
- [Using environment variables](#option-2-read-credentials-from-environment-variables)
- [Using credentials file](#option-3-read-credentials-from-kagglejson)
```

**Review by rosbo — APPROVED** [review]

**Pull request merged**